kong/plugins/prometheus/exporter.lua

```diff
@@ -168,6 +174,69 @@ local function collect()
                  "/metrics endpoint: ", err)
   end
 
+  -- erase all target/upstream metrics, prevent exposing old metrics
+  metrics.upstream_target_health:reset()
```

Comment:
@fffonion Will this flush the counters present in LuaVMs of other workers?

Reply:
those metrics are gauges andwe don't apply the optimization to gauges since they shouldn't be set in each request. so `reset` will work directly with shdict.